pkg/controller/flow_test.go

```diff
@@ -0,0 +1,182 @@
+/*
```

Comment:
IMO the filename is too generic. Could u precise the `flow_test`? it's scoped for binding only if yes, then what kind of flow u test? The `binding_auth_flow_test.go` will be ok name?

Reply:
I've renamed it to controller_flow_test. The best name would be jsut controller_test but it is used by existing unit tests.